### PR TITLE
Add glossary for typically misused terms

### PR DESCRIPTION
**Pull request by z1c0**

### CODE REVIEW

File: specification/glossary.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Glossary
+
+Below are a list of some OpenTelemetry specific terms that are used across this
+specification.
+
+## (Telemetry) Library
```
Comment (dyladan) [inline specification/glossary.md]: I think it will be very easy to confuse "Telemetry Library" with "Instrumenting Library." Is there some way we can make this naming less ambiguous?

File: specification/glossary.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Denotes the actual *OpenTelemetry* library in use.  
+See [Library](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md#library)
+
+<a name="instrumented_library"></a>
+
+## Instrumented Library
+
+Denotes the library instrumented with OpenTelemetry API calls, thus the library
```
Comment (dyladan) [inline specification/glossary.md]: This definition sounds a lot like the definition for the "instrumenting library." I would rather see something along the lines of "Denotes the library which is being monitored/observed." In many cases, this library will not be the one with the API calls, because it is monkey patched.

File: specification/glossary.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+the [Instrumented Library](#instrumented_library)). Hence, *Instrumented Library*
+and *Instrumenting Library* can be identical, but they don't necessarily have to.
+
+Example: `io.opentelemetry.contrib.mongodb`.
+
+Synonyms: *Instrumentation Library*, *Integration*.
+
+<a name="name"></a>
+
+## Name (of a Tracer or Meter)
```
Comment (dyladan) [inline specification/glossary.md]:
```suggestion
## Tracer Name / Meter Name
```

Comment (dyladan) [reply specification/glossary.md]:
```suggestion
## Tracer/Meter Name
```

File: specification/glossary.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+This refers to the `name` specified when creating a new `Tracer` or `Meter`
```
Comment (dyladan) [inline specification/glossary.md]: No mention of version?

File: specification/glossary.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+(see [Obtaining a Tracer](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#obtaining-a-tracer)/[Obtaining a Meter](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#obtaining-a-meter)). It is identical to
+[Instrumenting Library](#instrumenting_library).
+
+## Namespace
+
+This term applies to [Metric names](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#metric-names) only and is used to disambiguate identical metric
+names used in different libraries. The [Name](#name) provided for creating a `Meter`
```
Comment (dyladan) [inline specification/glossary.md]: If you add a mention of version to the Meter Name section above, it will probably be helpful to mention that the namespace is only the `name` part of the name-version pair.

File: specification/glossary.md (hunk 2 above)
Comment (z1c0) [reply specification/glossary.md]: Agreed. Let me rephrase that.

File: specification/glossary.md (hunk 1 above)
Comment (z1c0) [reply specification/glossary.md]: I know these are ambiguous, hence the glossary ;-)
I'm happy about suggestions though.

File: specification/glossary.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+## Tracer Name / Meter Name
+
+This refers to the `name` specified when creating a new `Tracer` or `Meter`
+(see [Obtaining a Tracer](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#obtaining-a-tracer)/[Obtaining a Meter](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#obtaining-a-meter)). It is identical to
+[Instrumenting Library](#instrumenting_library).
+
+## Namespace
+
+This term applies to [Metric names](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#metric-names) only and is used to disambiguate identical metric
+names used in different libraries. The [Name](#name) provided for creating a `Meter`
```
Comment (thisthat) [inline specification/glossary.md]: Since we have 3 different "library types", I would add the "instrumented" word.

File: specification/glossary.md (hunk 1 above)
Comment (arminru) [reply specification/glossary.md]: Yep, that's the name currently used in the project.

File: specification/glossary.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
+# Glossary
+
+Below are a list of some OpenTelemetry specific terms that are used across this
+specification.
+
+## (Telemetry) Library
+
+Denotes the actual *OpenTelemetry* library in use.
```
Comment (arminru) [inline specification/glossary.md]: Doesn't have to be OpenTelemetry actually, it could also be a compatible library producing data according to the OTel data format.

Comment (arminru) [inline specification/glossary.md]: Is this meant to be the API or SDK?

File: specification/glossary.md (hunk 2 above)
Comment (arminru) [reply specification/glossary.md]: "The library that is producing telemetry data" sounds to me almost certainly like the instrumenting library. The instrumented library is the one about which the data is produced imho.

File: specification/glossary.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+See [Library](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md#library)
+
+<a name="instrumented_library"></a>
+
+## Instrumented Library
+
+Denotes the library instrumented with OpenTelemetry API calls, thus the library
+that is producing telemetry data.
+
+Example: `org.mongodb`.
```
Comment (arminru) [inline specification/glossary.md]:
```suggestion
Example: `org.mongodb.client`.
```
to make it a bit clearer

File: specification/glossary.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+<a name="instrumenting_library"></a>
+
+## Instrumenting Library
+
+Denotes the library which provides the instrumentation *for* a given library (i.e.
+the [Instrumented Library](#instrumented_library)). Hence, *Instrumented Library*
+and *Instrumenting Library* can be identical, but they don't necessarily have to.
```
Comment (arminru) [inline specification/glossary.md]: ... can be identical if a library has OpenTelemetry instrumentation built-in / out-of-the-box

File: specification/glossary.md (hunk 6 above)
Comment (arminru) [reply specification/glossary.md]: nooo, "instrumenting"!!! 🙈

File: specification/glossary.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+This refers to the `name` specified when creating a new `Tracer` or `Meter`
+(see [Obtaining a Tracer](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#obtaining-a-tracer)/[Obtaining a Meter](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#obtaining-a-meter)). It is identical to
+[Instrumenting Library](#instrumenting_library).
+
+## Namespace
+
+This term applies to [Metric names](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#metric-names) only and is used to disambiguate identical metric
+names used in different libraries. The [Name](#name) provided for creating a `Meter`
+also serves as its namespace.
```
Comment (arminru) [inline specification/glossary.md]:
```suggestion
serves as its namespace.
```
otherwise it could be interpreted as if there was another, alternative namespace

File: specification/glossary.md (hunk 6 above)
Comment (thisthat) [reply specification/glossary.md]: :man_facepalming:

Comment (arminru) [reply specification/glossary.md]: no worries, that's why we'll get a glossary now 😉

Comment (thisthat) [reply specification/glossary.md]: At least it makes my comment more valid than before :sweat_smile:

File: specification/glossary.md (hunk 7 above)
Comment (dyladan) [reply specification/glossary.md]: It is my feeling that this is the SDK. The API should be the same for all vendors, but the SDK which is plugged will be different.

Comment (arminru) [reply specification/glossary.md]: Could be a fork of the API as well, or a compatible alternative to the standard API.

Comment (dyladan) [reply specification/glossary.md]: If you fork the API, you will lose the main (planned) advantage of opentelemetry which is that modules will use the API to instrument themselves. Any module which uses the official version of the API will not report data to your forked version.

Comment (z1c0) [reply specification/glossary.md]: Those questions regarding API vs SDK should be answered in the part of the specification I linked too.
@arminru already created an issue (https://github.com/open-telemetry/opentelemetry-specification/issues/484) for clearing that up.

File: specification/glossary.md (hunk 10 above)
Comment (z1c0) [reply specification/glossary.md]: "Also" refers to the semantics of `name` described in the context before. I'll try to make that clearer.

File: specification/glossary.md [hunk 11]
```diff
@@ -33,12 +36,15 @@ Synonyms: *Instrumentation Library*, *Integration*.
 
 ## Tracer Name / Meter Name
 
-This refers to the `name` specified when creating a new `Tracer` or `Meter`
-(see [Obtaining a Tracer](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#obtaining-a-tracer)/[Obtaining a Meter](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#obtaining-a-meter)). It is identical to
-[Instrumenting Library](#instrumenting_library).
+This refers to the `name` and (optional) `version` arguments specified when
+creating a new `Tracer` or `Meter` (see [Obtaining a Tracer](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#obtaining-a-tracer)/[Obtaining a Meter](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#obtaining-a-meter)). It is identical to [Instrumenting Library](#instrumenting_library).
```
Comment (arminru) [inline specification/glossary.md]: Please use relative links (they are slightly less likely to break).

File: specification/glossary.md [hunk 12]
```diff
@@ -33,12 +36,15 @@ Synonyms: *Instrumentation Library*, *Integration*.
...
 
 ## Namespace
 
 This term applies to [Metric names](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#metric-names) only and is used to disambiguate identical metric
-names used in different libraries. The [Name](#name) provided for creating a `Meter`
-also serves as its namespace.
+names used in different instrumenting libraries. The [Name](#name) provided
+for creating a `Meter` serves as its namespace in addition to the primary semantics
+describe [here](#name).
```
Comment (arminru) [inline specification/glossary.md]:
```suggestion
described [here](#name).
```

File: specification/glossary.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+Denotes the actual *OpenTelemetry* (or API compatible) library in use.  
+See [Library](data-resource-semantic-conventions.md#library)
+
+<a name="instrumented_library"></a>
+
+## Instrumented Library
+
+Denotes the library whose "actions" (e.g. called methods/functions, operations, ...)
+result in OpenTelemetry signals (traces, metrics, logs).
```
Comment (Oberon00) [inline specification/glossary.md]: This is confusing to me. The only function call that results in a trace for example is `StartSpan` and the library that calls StartSpan is the instrument**ing** library. I see the next sentence clarifies that, but still.

Suggestion: "The library of interest that signals are emitted for (traces, metrics, logs). The library that is monitored."

File: specification/glossary.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+The actual OpenTelemetry API calls can happen inside this library but might as well
+happen in a different one (the [Instrumenting Library](#instrumenting_library)).
+
+Example: `org.mongodb.client`.
+
+<a name="instrumenting_library"></a>
+
+## Instrumenting Library
+
+Denotes the library which provides the instrumentation *for* a given library (i.e.
```
Comment (Oberon00) [inline specification/glossary.md]:
```suggestion
The library which provides the instrumentation for a given library (i.e.
```

File: specification/glossary.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+the [Instrumented Library](#instrumented_library)). Hence, *Instrumented Library*
```
Comment (Oberon00) [inline specification/glossary.md]: I would leave out the part in parantheses, it is not 100% clear that it applies to the "given library" and not the "library which provides".

File: specification/glossary.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+Example: `io.opentelemetry.contrib.mongodb`.
+
+Synonyms: *Instrumentation Library*, *Integration*.
+
+<a name="name"></a>
+
+## Tracer Name / Meter Name
+
+This refers to the `name` and (optional) `version` arguments specified when
```
Comment (Oberon00) [inline specification/glossary.md]: Does it really refer to the version too? I think we have to concede that when someone is using "tracer/meter name" it is unclear whether the version is included.

File: specification/glossary.md [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+creating a new `Tracer` or `Meter` (see [Obtaining a Tracer](api-tracing.md#obtaining-a-tracer)/[Obtaining a Meter](api-metrics-user.md#obtaining-a-meter)). It is identical to [Instrumenting Library](#instrumenting_library).
```
Comment (Oberon00) [inline specification/glossary.md]:
```suggestion
creating a new `Tracer` or `Meter` (see [Obtaining a Tracer](api-tracing.md#obtaining-a-tracer)/[Obtaining a Meter](api-metrics-user.md#obtaining-a-meter)). It identifies the [Instrumenting Library](#instrumenting_library).
```

File: specification/glossary.md [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+
+## Namespace
+
+This term applies to [Metric names](api-metrics-user.md#metric-names) only and is used to disambiguate identical metric
```
Comment (Oberon00) [inline specification/glossary.md]:
```suggestion
This term applies to [Metric names](api-metrics-user.md#metric-names) only. The namespace is used to disambiguate identical metric
```

File: specification/glossary.md [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+names used in different instrumenting libraries. The [Name](#name) provided
```
Comment (Oberon00) [inline specification/glossary.md]:
```suggestion
names used in different [instrumenting libraries](#instrumenting_library). The [Name](#name) provided
```

File: specification/glossary.md [hunk 20]
```diff
@@ -0,0 +1,50 @@
...
+for creating a `Meter` serves as its namespace in addition to the primary semantics
+described [here](#name).
+
+The `version` argument is not relevant here and will not be included in
```
Comment (Oberon00) [inline specification/glossary.md]: I think this is already too much detail for a glossary.

Comment (z1c0) [reply specification/glossary.md]: Leaving that in, since it was demanded by previous reviewers and upvoted (liked).

File: specification/glossary.md (hunk 16 above)
Comment (z1c0) [reply specification/glossary.md]: Leaving that in, since it was demanded by previous reviewers and upvoted (liked).

File: specification/glossary.md [hunk 21]
```diff
@@ -0,0 +1,50 @@
+# Glossary
+
+Below are a list of some OpenTelemetry specific terms that are used across this
+specification.
+
+## (Telemetry) Library
```
[the remark this answers is not in the document]
Comment (z1c0) [reply specification/glossary.md]: It does not say **Telemetry** Library in the referenced part of the spec, so I deliberately put the parentheses there.

File: specification/glossary.md (hunk 14 above)
Comment (z1c0) [reply specification/glossary.md]: Honestly, I like my wording better and also the emphasis on *for*.

**Comment by Oberon00** [comment]
https://github.com/dynatrace-oss-contrib/opentelemetry-specification/pull/10#discussion_r385607656 is hidden by github and unresolved.

File: specification/glossary.md (hunk 15 above)
Comment (z1c0) [reply specification/glossary.md]: Using *for* again should clear that up.
